lib/hooks/typescript-compilation.ts

```diff
@@ -25,9 +25,11 @@ fiberBootstrap.run(() => {
 
 		if ($fs.exists(pathToTsConfig).wait()) {
 			let json = $fs.readJson(pathToTsConfig).wait();
-			typeScriptCompilationService.compileWithDefaultOptions({ noEmitOnError: !!(json && json.compilerOptions && json.compilerOptions.noEmitOnError) }).wait();
+			let noEmitOnError = !!(json && json.compilerOptions && json.compilerOptions.noEmitOnError);
+			typeScriptService.transpileWithDefaultOptions(project.getProjectDir().wait(), { compilerOptions: { noEmitOnError } }).wait();
 		} else {
-			typeScriptCompilationService.compileFiles({ noEmitOnError: false }, typeScriptFiles.typeScriptFiles, typeScriptFiles.definitionFiles).wait();
+			let pathToDefaultDefinitionFiles = path.join(__dirname, "../../resources/typescript-definitions-files");
```

Comment:
You can use the `$resource` module's `resolvePath` method to avoid having to `path.join` multiple `..`s